roles/cs.nginx-https-termination/tasks/000-facts.yml

```diff
@@ -0,0 +1,149 @@
+- name: Compute basic global config
+  set_fact:
+    https_termination_default_upstream_uri: "{{ https_termination_upstream_scheme }}://{{ https_termination_upstream_host }}:{{ https_termination_upstream_port }}"
+    https_termination_nginx_install_global_configs:
+      - label: Specialized JSON log for debugging cache misses
+        src: nginx.cache-miss-log.global.conf
+        dst: "{{ https_termination_conf_cache_miss_log_path }}"
+
+      - label: Global configuration maps for server cookie rewrites
+        src: nginx.cookie-rewrite.global.conf
+        dst: "{{ https_termination_conf_cookie_rewrite_path }}"
+
+
+- name: Compute primary runtime vhost configuration
```

Comment:
The next three tasks just prepare vars for every vhost in one place. It needs to be done in 3 steps because some vars are dependent on others and ansible cannot self-reference them in a single step.

For example the resulting vhost configuration will look something like this:
```yaml
  https_termination_vhosts:
    magesuite-proto.magesuite.net:
      conf_custom_block: false
      conf_fragment_dir: /etc/nginx/server.d/magesuite-proto.magesuite.net
      conf_path: /etc/nginx/conf.d/050-vhost-magesuite-proto.magesuite.net.conf
      crt_acme_archive_dir: /etc/letsencrypt/archive/magesuite-proto.magesuite.net
      crt_acme_live_dir: /etc/letsencrypt/live/magesuite-proto.magesuite.net
      crt_acme_live_fullchain_path: /etc/letsencrypt/live/magesuite-proto.magesuite.net/fullchain.pem
      crt_acme_live_key_path: /etc/letsencrypt/live/magesuite-proto.magesuite.net/privkey.pem
      crt_acme_renewal_conf_path: /etc/letsencrypt/renewal/magesuite-proto.magesuite.net.conf
      crt_acme_webroot_dir: /etc/letsencrypt/webroot/magesuite-proto.magesuite.net
      crt_archive_dir: /etc/nginx/certs/archive/magesuite-proto.magesuite.net
      crt_boot_cert_path: /etc/nginx/certs/archive/magesuite-proto.magesuite.net/selfsigned.cert.pem
      crt_boot_csr_path: /etc/nginx/certs/archive/magesuite-proto.magesuite.net/selfsigned.csr.pem
      crt_boot_fullchain_path: /etc/nginx/certs/archive/magesuite-proto.magesuite.net/selfsigned.fullchain.pem
      crt_boot_key_path: /etc/nginx/certs/archive/magesuite-proto.magesuite.net/selfsigned.privkey.pem
      crt_do_acme: true
      crt_do_bootstrap: true
      crt_do_install: false
      crt_install_cert_path: /etc/nginx/certs/archive/magesuite-proto.magesuite.net/installed.cert.pem
      crt_install_cert_src: false
      crt_install_chain_path: /etc/nginx/certs/archive/magesuite-proto.magesuite.net/installed.chain.pem
      crt_install_chain_src: false
      crt_install_fullchain_path: /etc/nginx/certs/archive/magesuite-proto.magesuite.net/installed.fullchain.pem
      crt_install_key_path: /etc/nginx/certs/archive/magesuite-proto.magesuite.net/installed.privkey.pem
      crt_install_key_src: false
      crt_live_dir: /etc/nginx/certs/live
      crt_live_fullchain_path: /etc/nginx/certs/live/magesuite-proto.magesuite.net.fullchain.pem
      crt_live_key_path: /etc/nginx/certs/live/magesuite-proto.magesuite.net.privkey.pem
      label: magesuite-proto.magesuite.net
      name: magesuite-proto.magesuite.net
      params:
        letsencrypt: true
        name: magesuite-proto.magesuite.net
        server_name: magesuite-proto.magesuite.net
      safe_name: magesuite_proto_magesuite_net
      server_aliases: []
      server_aliases_redirect: false
      server_custom_block: false
      server_name: magesuite-proto.magesuite.net
      server_redirect_path_code: '301'
      server_redirect_path_mapping: {}
      upstream:
        fragment: ''
        hostname: 127.0.0.1
        netloc: 127.0.0.1:8000
        password: null
        path: ''
        port: 8000
        query: ''
        scheme: http
        username: null
      upstream_proxy_http: true
      upstream_uri: http://127.0.0.1:8000
```